### DISCUSSION TITLE
Cypress Headless Mode Recording

### DISCUSSION THREAD

**Discussion by alyoung1621** [cypress-io/cypress#28464] [Questions and Help]
When running e2e tests in headless mode (`cypress run`) with `video: true` in `cypress.config.ts`, are the videos also uploaded to Cypress Cloud or only saved to `cypress/videos` folders? I don't have an account/am not logged in to Cypress Cloud.

**Comment by portyu9** [comment]
Hi @alyoung1621 

With:

```js
video: true
```

and a normal:

```bash
cypress run
```

the videos are stored **locally** in Cypress's `videosFolder`, which defaults to:

```text
cypress/videos
```

They are not automatically uploaded to Cypress Cloud just because video recording is enabled.

Cloud upload happens when you run the tests as a recorded Cypress Cloud run, for example with:

```bash
cypress run --record
```

So if your project is not configured for Cypress Cloud and you are not running with `--record`, the video remains local.

In short:

- `video: true` → record a local video.
- `--record` + Cypress Cloud project → upload run data and videos to Cypress Cloud.

Please mark this answer as accepted if it helped, thank you.
